proposals/sycl_modules.md

```diff
@@ -707,66 +738,186 @@ value to the specialization constant.
 The program can then be build with `build_with_kernel_type` to create a 
 specialized kernel.
 
+### Specialization constant types
+
 On the new, type-safe, module approach, specialization constants are 
-associated with the OpenCL backend. 
-Other backends (e.g. Vulkan) may offer the same functionality, so the
-specialization constant type is defined as an (experimental) general 
-SYCL type as below:
+typically associated with SPIR-V module images but are not limited to such image.
+The specialization constant type is defined as an general SYCL type as below:
 
 ```cpp
 namespace sycl {
-namespace experimental {
 
-template <typename T, typename ID = T>
-class spec_constant {
+template <typename T>
+class spec_id {
 private:
   // Implementation defined constructor.
-  spec_constant(/* Implementation defined */);
+  spec_id(const spec_id&) = delete;
+  spec_id(spec_id&&) = delete;
+
+
 public:
+
+  using type = T;
+
+  // Argument `Args` are forwarded to an underlying T Ctor.
+  // This allow the user to setup a default value for the spec_id instance.
+  // The initialization of T must be evaluated at compile time to be valid.
+  template<class... Args >
+  explicit constexpr spec_id(Args&&...);
+};
+
+template <class T, spec_id<T>& s>
+class spec_constant {
+public:
+  using type = T;
+
+  // Create an empty (invalid) spec constant.
+  // Remains invalid until initialized by a call
+  // to handler::set_spec_constant or
+  // to handler::get_spec_constant
   spec_constant();
 
+  spec_constant(const spec_constant&) = default;
+  spec_constant& operator=(const spec_constant&) = default;
+
   T get() const; // explicit access.
   operator T() const;  // implicit conversion.
 };
 
-}  // namespace experimental
 }  // namespace sycl
 ```
 
+`spec_id` is used by the *SYCL implementation* to *reference* constant whose
+concrete value will only be known during the application execution.
+To create a new `spec_id` in a module, the user creates a `spec_id` object with an *automatic* or *static* storage duration in the namespace or class scope.
+The user then uses references to this object to bind the `spec_constant` value to the *identifier*.
+
+`spec_id` objects must be forward declare-able and cannot be moved or copied.
```

Comment:
A `spec_id` instance is an identifier. The sole purpose of instances of this class is to bind kernel constants with the SYCL runtime so it can manipulate it.

what would be the meaning of moving/copy an identifier ?

Reply:
What is the rational of *not* moving/copy an identifier ?
A typical OpenCL id in C is copyable or moveable for example.

Reply:
OK, now I realize that it is not like a typical OpenCL id. It is just the address of the object you are using.